### PR TITLE
Remove WorkOSBaseResource

### PR DESCRIPTION
**Pull request by mattgd**
## Description
Remove WorkOSBaseResource and refactor test mocks.

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[ ] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: tests/test_sso.py [hunk 1]
```diff
@@ -15,54 +16,37 @@
...
+            email="demo@workos-magic-link.com",
+            organization_id=None,
+            connection_id="conn_01EMH8WAK20T42N2NBMNBCYHAG",
+            connection_type="MagicLink",
+            idp_id="",
+            first_name=None,
+            last_name=None,
+            groups=None,
+            raw_attributes={},
+        ).dict()
```
Comment (mattgd) [inline tests/test_sso.py:35]: Haven't quite landed on the perfect scheme for these mocks. We could add additional flexibility by allowing all parameters to be passed, but provide mock defaults.

Comment (tribble) [reply tests/test_sso.py:35]: Sounds a lot like a test fixture generation tool, like Ruby's machinist or FactoryGirl. A problem for future us.

One advantage of reusing our existing pydantic models as fixtures is it gives us the ability to include fields in the test response data that are not part of the pydantic model.

File: workos/resources/list.py [hunk 2]
```diff
@@ -24,102 +23,11 @@
 from workos.resources.mfa import AuthenticationFactor
 from workos.resources.organizations import Organization
 from pydantic import BaseModel, Field
-from workos.resources.sso import Connection, ConnectionWithDomains
+from workos.resources.sso import ConnectionWithDomains
 from workos.resources.user_management import Invitation, OrganizationMembership, User
 from workos.resources.workos_model import WorkOSModel
 
 
-class WorkOSListResource(WorkOSBaseResource):
```
Comment (tribble) [inline workos/resources/list.py:32]: See Ya

**Review by tribble — APPROVED** [review]
Happy Dog

**Pull request merged**